worlds/metroidprime/config.py

```diff
@@ -1,1811 +1,1639 @@
-                    }
-                }
-            }
-        }
-    }
-    return config
+from typing import Dict, Any
+
+
+def starting_inventory(world, item) -> bool:
+    items = world.multiworld.precollected_items.values()
+    if item in items:
+        return True
+    else:
+        return False
+
+
+def starting_ammo(world, item) -> int:
+    items = world.multiworld.precollected_items.values()
+    count = 0
+    if item == "Missile Expansion":
+        for i in items:
+            if i == "Missile Expansion" or i == "Missile Launcher":
+                count += 5
+        return count
+    if item == "Energy Tank":
+        for i in items:
+            if i == "Energy Tank":
+                count += 1
+    if item == "Power Bomb":
+        for i in items:
+            if i == "Power Bomb" or i == "Power Bomb Expansion":
+                count += 1
+    return count
+
+
+def spring_check(spring) -> bool:
+    if spring == 1:
+        return True
+    else:
+        return False
+
+
+def ridley(boss) -> bool:
+    if boss == 0 or boss == 1:
+        return False
+    else:
+        return True
+
+
+def temple_dest(boss) -> str:
+    if boss == 0 or boss == 2:
+        return "Crater Entry Point"
+    else:
+        return "Credits"
+
+
+def item_text(world, location) -> str:
+    loc = world.multiworld.get_location(location, world.player)
+    player_name = f"{world.multiworld.player_name[location.player]} " if loc.player != world.player else ""
+    return f"{player_name}{loc.item.name}"
+
+
+def item_model(world, location) -> str:
+    loc = world.multiworld.get_location(location, world.player)
+    if loc.native_item:
+        nam = loc.item.name
+        if nam == "Missile Expansion" or nam == "Main Missile":
+            return "Missile"
+        else:
+            return nam
+    else:
+        return "Nothing"
+
+
+def make_config(world):
+    options = world.options
+    config = {
+        "$schema": "https://randovania.org/randomprime/randomprime.schema.json",
+        "inputIso": "prime.iso",
+        "outputIso": "prime_out.iso",
+        "forceVanillaLayout": False,
+        "preferences": {
+            "forceFusion": False,
+            "cacheDir": "cache",
+            "qolGeneral": True,
+            "qolGameBreaking": False,
+            "qolCosmetic": True,
+            "qolCutscenes": "Skippable",
+            "qolPickupScans": True,
+            "mapDefaultState": "Always",
+            "artifactHintBehavior": "All",
+        },
+        "gameConfig": {
+            "startingRoom": "Tallon Overworld:Landing Site",
+            "springBall": spring_check(options.spring_ball.value),
+            "warpToStart": True,
+            "nonvariaHeatDamage": False,
+            "staggeredSuitDamage": False,
+            "heatDamagePerSec": 10.0,
+            "poisonDamagePerSec": 0.11,
+            "phazonDamagePerSec": 0.964,
+            "phazonDamageModifier": "Default",
+            "autoEnabledElevators": True,

```

Comment:
I turned this on for testing, let me know if you want to keep it disabled or we can add it as a config option 👍